diff --git a/src/hexcurse.c b/src/hexcurse.c
index 9723c2f..090c924 100644
--- a/src/hexcurse.c
+++ b/src/hexcurse.c
@@ -24,6 +24,7 @@
 \******************************************************************************/
 
 #include "hex.h"					/* custom header      */
+#include "zos.h"
 
 /*#define DEBUG_LLIST*/
 /*#define DEBUG_GOTO*/
@@ -264,6 +265,9 @@ off_t parseArgs(int argc, char *argv[])
     } else if (fpINfilename && strcmp(fpINfilename, "")) {
         if ((fpIN = fopen(fpINfilename, "r")) == NULL)
             exit_err("Could not open file");
+#ifdef __MVS__
+    __disableautocvt(fileno(fpIN));
+#endif
     }
 
     return ((fpIN != NULL) ? maxLoc(fpIN):0);		/* return file length */
