diff --git a/src/hexcurse.c b/src/hexcurse.c
index 9723c2f..4e8c8fa 100644
--- a/src/hexcurse.c
+++ b/src/hexcurse.c
@@ -102,7 +102,7 @@ int main(int argc, char *argv[])			/* main program       */
     printHex = TRUE;							/* address format     */
     USE_EBCDIC = FALSE;							/*use ascii by default*/
     TERM_COLORS = FALSE;                     /*don't use term defined colors by default */
-    FNUMBERS = FALSE;     /* keep old behaviour without function key numbers by default */
+    FNUMBERS = TRUE;     /* keep old behaviour without function key numbers by default */
 
 							/* get cmd line args  */
     len = parseArgs(argc, argv);
